src/lib/control_allocation/control_allocation/CMakeLists.txt

```diff
@@ -34,6 +34,8 @@
 px4_add_library(ControlAllocation
 	ControlAllocation.cpp
 	ControlAllocation.hpp
+	ControlAllocationMetric.cpp
```

Comment:
Did you consider calling it ControlAllocationPseudoInverseMetric? That would make it more clear which parts of the existing, normalized, pipeline you're replacing. 

Next step would then be to move all the normalization and scaling of the allocation matrix out of the PseudoInverse. The algorithms in the end don't care whether the allocation matrix is normalized or not, do they? Would ControlAllocationSequentialDesaturation work out of the box with a metric allocation matrix?

Reply:
To address this I've created an internal variable to activate metric allocation. Then we keep the same structure and if non-metric is being used, the allocator behaves as previously expected. 

I'd need to look in detail towards the sequential desaturation algorithm, but since I don't use it, I don't have a good answer as of now. In principle, yes, this should be possible.